mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala

```diff
@@ -0,0 +1,238 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.regression
+
+import java.io.Serializable
+import java.util.Arrays.binarySearch
+
+import org.apache.spark.api.java.{JavaDoubleRDD, JavaRDD}
+import org.apache.spark.rdd.RDD
+
+/**
+ * Regression model for Isotonic regression
+ *
+ * @param features Array of features.
+ * @param labels Array of labels associated to the features at the same index.
+ */
+class IsotonicRegressionModel (
+    features: Array[Double],
+    val labels: Array[Double])
+  extends Serializable {
+
+  /**
```

Comment:
I have added validations for `features` (now `boundaries`) to be ordered and for `boundaries` and `labels` to have the same length.

Could not find an existing solution for order validation, so used the fastest available (see https://groups.google.com/forum/#!topic/scala-user/oo4nuA_3U7Q), because genericity was not required.